### PR TITLE
fixes for auxhist output

### PR DESCRIPTION
**Pull request by mvertens**
### Description of changes
fixes for auxhist output

### Specific notes
These are fixes to get the auxiliary output working for cesm spinup runs.

Contributors other than yourself, if any:

CMEPS Issues Fixed (include github issue #):

Are changes expected to change answers? (specify if bfb, different at roundoff, more substantial) 

Any User Interface Changes (namelist or namelist defaults changes)?

### Testing performed

Testing performed if application target is CESM:
- [ ] (recommended) CIME_DRIVER=nuopc scripts_regression_tests.py
   - machines:
   - details (e.g. failed tests):
- [ ] (recommended) CESM testlist_drv.xml
   - machines and compilers:
   - details (e.g. failed tests):
- [ ] (optional) CESM prealpha test
   - machines and compilers
   - details (e.g. failed tests):
- [ ] (other) please described in detail
   - machines and compilers
   - details (e.g. failed tests):

Testing performed if application target is UFS-coupled:
- [ ] (recommended) UFS-coupled testing
   - description:
   - details (e.g. failed tests):

Testing performed if application target is UFS-HAFS:
- [ ] (recommended) UFS-HAFS testing
   - description:
   - details (e.g. failed tests):

### Hashes used for testing:

- [ ] CESM:
  - repository to check out: https://github.com/ESCOMP/CESM.git
  - branch/hash:
- [ ] UFS-coupled, then umbrella repostiory to check out and associated hash:
  - repository to check out:
  - branch/hash:
- [ ] UFS-HAFS, then umbrella repostiory to check out and associated hash:
  - repository to check out:
  - branch/hash:

### CODE REVIEW

**Comment by mvertens** [comment]
@fischer- could you please use this branch in alpha10c and try to see if the prealpha tests are bfb?
I'm running out of time to get all of these changes in.

File: mediator/med_phases_history_mod.F90 [hunk 1]
```diff
@@ -1058,6 +1059,10 @@ subroutine med_phases_history_write_comp_aux(gcomp, compid, auxcomp, rc)
     integer                 :: yr,mon,day,sec    ! time units
     real(r8)                :: time_val          ! time coordinate output
     real(r8)                :: time_bnds(2)      ! time bounds output
+    !DEBUG
```
Comment (jedwards4b) [inline mediator/med_phases_history_mod.F90:1062]: Should this code be removed?

**Review by jedwards4b — CHANGES_REQUESTED** [review]

File: mediator/med_phases_history_mod.F90 (hunk 1 above)
Comment (mvertens) [reply mediator/med_phases_history_mod.F90:1062]: @jedwards4b - yes - could you please remove it.

File: mediator/med_methods_mod.F90 [hunk 2]
```diff
@@ -445,8 +439,10 @@ subroutine med_methods_FB_init(FBout, flds_scalar_name, fieldNameList, FBgeom, S
 
              ! ungridded dimensions might be present in the input states or field bundles
              if (present(FBflds)) then
-                call med_methods_FB_getFieldN(FBflds, n, lfield, rc=rc)
+                call ESMF_FieldBundleGet(FBflds, fieldName=lfieldnamelist(n), field=lfield, rc=rc)
                 if (chkerr(rc,__LINE__,u_FILE_u)) return
+                ! call med_methods_FB_getFieldN(FBflds, n, lfield, rc=rc)
```
[the remark this answers is not in the document]
Comment (mvertens) [reply mediator/med_methods_mod.F90:444]: I think so.

**Review by jedwards4b — APPROVED** [review]

**Comment by DeniseWorthen** [comment]
I tested in UFS and everything passed.

**Review by DeniseWorthen — APPROVED** [review]

**Review by uturuncoglu — APPROVED** [review]

**Comment by mvertens** [comment]
@denise, @uturuncoglu, @jedwards4b - thanks so much for your quick response to this PR.

**Pull request merged**